crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -326,7 +326,7 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                             match err.status_code {
                                 // marking failure for 2xx because this is genuine payment failure
                                 200..=299 => storage::enums::AttemptStatus::Failure,
-                                _ => payment_data.payment_attempt.status,
+                                _ => router_data.status,
```

Comment:
why this change is required?

Reply:
Connector doesn't have access to payment_attempt.status , we need to update it using router_data